assets/js/utils/rtcUtils.ts

```diff
@@ -294,22 +307,50 @@ const filterDevices = (allDevices: MediaDeviceInfo[], type: string) => {
   return allDevices.filter((device) => device.deviceId != "default" && device.kind == type);
 };
 
-const addOrReplaceTrack = (
+export const addOrReplaceTrack = (
   client: Client,
   track: MediaStreamTrack,
-  playerCallback: (sourceType: SourceType) => void
+  setPresenterStreams: React.Dispatch<React.SetStateAction<PresenterStreams>>
 ) => {
-  if (!presenterStreams[client.email]) presenterStreams[client.email] = new MediaStream();
-  const curTrack = findTrackByType(client, track.kind as SourceType);
+  setPresenterStreams((prev) => {
+    const presenterStreams = { ...prev };
+    if (!presenterStreams[client.email])
+      presenterStreams[client.email] = {
+        name: client.name,
+        email: client.email,
+        stream: new MediaStream(),
+      };
```

Comment:
You are doing the same thing in `onJoinSuccess` I would remove one of this occurence.